spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1836,6 +1838,48 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
           }
         }
 
+      case join: ShuffledHashJoinExec if isCometOperatorEnabled(op.conf, "hash_join") =>
+        if (join.buildSide == BuildRight) {
+          // DataFusion HashJoin assumes build side is always left.
+          // TODO: support BuildRight
```

Comment:
do we wanna support BuildRight in Datafusion side or we wanna do some thing smart here 

like this : https://github.com/apache/spark/pull/29097/files
flipping the join sides for Innerlike joins ? 

@viirya wdyt ?

Reply:
This is when Spark planner decides to use right side as build side for HashJoin. I don't think we will go to do flop join sides in Comet. We may need to update DataFusion HashJoin to support right side as build side.

Reply:
ACK, this makes sense so it's like comet shouldn't make any decisions making changes on spark plan itself, and just focus on execution rather than planning :) !

Reply:
Just reading along, won't `BuildRight` be just swapping build and probe? I think the left/right naming in DataFusion is just there because it used to be called like this, but in practice there is only build vs probe in the operator.

Reply:
Yea, in DataFusion, only left side could be the build side. But in Spark, the HashJoin operator has a build side parameter to indicate which side is build side. The operator will do right thing accordingly internally. So currently we cannot just create a DataFusion HashJoin operator with right side as build side.

It can be swapped between left and right side, only if we also swap outputs and also column binding in joining keys and joining filter. I'd like to relax the build side constraint in DataFusion instead of doing the swap in Comet.

Reply:
Makes sense, thanks for explaining

Reply:
Created https://github.com/apache/arrow-datafusion/issues/9603 to track it.